spec/factories/author.rb

```diff
@@ -1,12 +1,19 @@
 FactoryBot.define do
   sequence(:random_id) do |n|
-    @random_ids ||= (10_000..1_000_000).to_a.shuffle
+    @random_ids ||= Array.new(10_000) { SecureRandom.random_number(990_000) + 10_000 }
```

Comment:
This factory is ridiculous.  It generates 10,000 random numbers to give back one.  Previously, it initialized a `990,000` member `Array` and shuffled it, which is **worse**!  The most authors our tests ever use is `5`.  

Why not just generate the ids on demand (per `n`), record each generated ids in a Hash (`@random_ids`) and regenerate if there is ever collision?  The only downside of this approach would be when the namespace started to fill up and we had collisions, but clearly our usage will never approach that!

Reply:
The only question with adopting a Hash solution is whether that could interfere with the design of the `sequence` method.  Originally, this was a cop-n-pasta solution and it works, although I also don't love it for these reasons ^^.  I'll do a bit more homework on that `sequence` method and see if a Hash solution works out OK.